### PR TITLE
[MM-20465] migrate single_image_view to typescript

### PR DESCRIPTION
**Pull request by Tak-Iwamoto**
#### Summary
migrate components/single_image_view to typescript
#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15475
JIRA: https://mattermost.atlassian.net/browse/MM-20465

### CODE REVIEW

**Comment by wiggin77** [comment]
@amyblais I think at least one front-end reviewer should look at this.  Currently there are two server devs assigned.

File: components/single_image_view/single_image_view.tsx [hunk 1]
```diff
@@ -16,25 +17,35 @@ import ViewImageModal from 'components/view_image';
...
+    dimensions: {
+        width: number;
+        height: number;
+    },
+}
 
+export default class SingleImageView extends React.PureComponent<Props, State> {
+    private mounted: boolean;
     static defaultProps = {
-        fileInfo: {},
```
Comment (nevyangelova) [inline components/single_image_view/single_image_view.tsx:32]: Are we sure the default value won't be used?

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: components/single_image_view/single_image_view.tsx (hunk 1 above)
Comment (wiggin77) [reply components/single_image_view/single_image_view.tsx:32]: Hi @Tak-Iwamoto, thanks for your contribution.  I would like to approve this PR but need the above question addressed first.  Thanks!

**Review by nevyangelova — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: components/single_image_view/single_image_view.tsx [hunk 2]
```diff
@@ -16,25 +17,39 @@ import ViewImageModal from 'components/view_image';
...
-        fileInfo: PropTypes.object.isRequired,
-        isRhsOpen: PropTypes.bool.isRequired,
-        compactDisplay: PropTypes.bool,
-        isEmbedVisible: PropTypes.bool,
-        actions: PropTypes.shape({
-            toggleEmbedVisibility: PropTypes.func.isRequired,
-        }).isRequired,
+type Props = {
+    post: Post;
+    fileInfo: FileInfo;
```
Comment (nevyangelova) [inline components/single_image_view/single_image_view.tsx:22]:
```suggestion
    fileInfo: FileInfo | undefined;
```

**Pull request merged**